kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java

```diff
@@ -446,19 +447,18 @@ TableInfo constructTableInfo(TableId table, com.google.cloud.bigquery.Schema big
     if (intermediateTables) {
       // Shameful hack: make the table ingestion time-partitioned here so that the _PARTITIONTIME
       // pseudocolumn can be queried to filter out rows that are still in the streaming buffer
-      builder.setTimePartitioning(TimePartitioning.of(Type.DAY));
+      setTimePartitioningForDefinition(table, builder);
     } else {
-      TimePartitioning timePartitioning = TimePartitioning.of(Type.DAY);
+      TimePartitioning timePartitioning = setTimePartitioningForDefinition(table, builder);
       if (timestampPartitionFieldName.isPresent()) {
         timePartitioning = timePartitioning.toBuilder().setField(timestampPartitionFieldName.get()).build();
       }
-  
-      builder.setTimePartitioning(timePartitioning);
 
+      builder.setTimePartitioning(timePartitioning);
```

Comment:
It feels odd to have a `setPartitioningForDefinition` table that sets the `TimePartitioning` [here](https://github.com/confluentinc/kafka-connect-bigquery/pull/61/files#diff-ca041c1c31ebc1ccb34783c9f1a727c172e94a985a0d82ec6ba39c3d487b73a4R488), returns the value and sets it here again. I'd rather rename the other method `getPartitioningForDefinition` that only returns the value, and then here you decide how you want to set it.